optimum/habana/transformers/models/llama/modeling_llama.py

```diff
@@ -772,6 +772,12 @@ def forward(
             batch_size, seq_length = inputs_embeds.shape[:2]
         else:
             raise ValueError("You have to specify either input_ids or inputs_embeds")
+        if hasattr(self.config, "use_fused_rope") and self.config.use_fused_rope is False:
+            global has_fused_rope
+            has_fused_rope = False
+        if hasattr(self.config, "use_fused_rms_norm") and self.config.use_fused_rms_norm is False:
+            global has_fused_rms_norm
+            has_fused_rms_norm = False
```

Comment:
has_fused_rms_norm also has some issues?

atleast in sft, dpo, ppo i see only use_fused_rope flag set to false

Reply:
ppo has backward issue in rms. do not see in dpo and sft.